File: ApplicationLLA.Server/Models/Post.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace ApplicationLLA.Server.Models
{
    public class Post
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int PostId { get; set; }

        [StringLength(60)]
        public required string Category { get; set; }

        [StringLength(120)]
        public required string Title { get; set; }

        [StringLength(2000)]
        public required string Description { get; set; }

        public int Price { get; set; }
        [StringLength(8)]
        [Required]
        public required string PriceCurrency {  get; set; }
        [StringLength(16)]
        [Required]
        public required string PriceWorkUnit { get; set; }

        [StringLength(30)]
        public required string City { get; set; }

        [StringLength(30)]
        public required string County { get; set; }

        public required bool IsPostActive { get; set; } = false;
        public required string WorkerId { get; set; }
        public virtual Worker? Worker { get; set; }
        public virtual List<Reservation>? Reservations { get; set; } = new List<Reservation>();

    }
}

File: frontend/src/Models/Post.ts
export type PostGet = {
    postId: number;
    workerId: string;
    workerName: string;
    pictureLink: string;
    category: string;
    title: string;
    description: string;
    price: number;
    priceCurrency: string,
    priceWorkUnit: string,
    city: string;
    county: string;
    isPostActive: boolean;
    userScore: number | null;
}
export type AddPost = {
    category: string;
    title: string;
    description: string;
    price: number;
    priceCurrency: string,
    priceWorkUnit: string,
}

export type WorkerPosts = {
    postId: number;
    workerId: string;
    category: string;
    title: string;
    description: string;
    price: number;
    priceCurrency: string,
    priceWorkUnit: string,
    city: string;
    county: string;
    countOfWaitingReservation: number;
    isPostActive: boolean;
    reservList: postReservations[];
}

export type postReservations = {
    reservationId: number;
    customerId: string;
    postId: number;
    status: string;
    reservationDate: string;
    reservationNote: string;
    customerPictureLink: string;
    postTitle: string;
    customerName: string;
}